stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/ZipCall.kt

```diff
@@ -2,88 +2,70 @@ package io.getstream.chat.android.client.call
 
 import io.getstream.chat.android.client.errors.ChatError
 import io.getstream.chat.android.client.utils.Result
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
 
 internal object ZipCall {
 
-    fun <A : Any, B : Any, C : Any> zip(
-        callA: Call<A>,
-        callB: Call<B>,
-        callC: Call<C>
-    ): Call<Triple<A, B, C>> {
-        return object : ChatCallImpl<Triple<A, B, C>>() {
-            override fun execute(): Result<Triple<A, B, C>> {
-                val resultA = callA.execute()
-
-                if (!resultA.isSuccess) return Result(null, resultA.error())
-                val resultB = callB.execute()
-                if (!resultB.isSuccess) return Result(null, resultB.error())
-                val resultC = callC.execute()
-                if (!resultC.isSuccess) return Result(null, resultC.error())
-
-                return Result(Triple(resultA.data(), resultB.data(), resultC.data()), null)
-            }
+    fun <A : Any, B : Any> zip(callA: Call<A>, callB: Call<B>): Call<Pair<A, B>> {
```

Comment:
nit: why function? You could also create a class with the same logic and name it ZipCall

Reply:
This was because of how the old implementation was structured, but good catch, I've changed this to match the style of `MapCall`.